lib/wallet/txdb.js

```diff
@@ -1706,6 +1707,12 @@ class TXDB {
         const nameHash = covenant.getHash(0);
         const prevout = tx.inputs[i].prevout;
 
+        if (reserved.has(nameHash)) {
```

Comment:
Will `reserved.has()` work still work after the 4 year claim period expires and users can win these names in auction? We get the whole TX in this funciton, does it have the view also? The best would be if we could test `isClaim()` on the input coin

Reply:
Oh right, we're inside the wallet, just get the coin from the wallet :-) 👍

Reply:
(nit: `if (!brv)` should work right?)

Reply:
Or maybe even this:
```js
case types.REGISTER: {
  assert(i < tx.inputs.length);

  const prevout = tx.inputs[i].prevout;
  const coin = await this.getCoin(prevout.hash, prevout.index);

  assert(coin.covenant.isReveal() || coin.covenant.isClaim());

  if (height === -1) {
    state.ulocked(path, -coin.value);
    state.ulocked(path, output.value);
  } else {
    state.clocked(path, -coin.value);
    state.clocked(path, output.value);
  }

  break;
}
```

Reply:
Ohhhh my yes.